renderdoc/driver/vulkan/vk_initstate.cpp

```diff
@@ -1975,28 +1975,78 @@ void WrappedVulkan::Apply_InitialState(WrappedVkRes *live, VkInitialContents ini
   }
   else if(type == eResDeviceMemory)
   {
+    Intervals<InitReqType> resetReq;
+    ResourceId orig = GetResourceManager()->GetOriginalID(id);
+    MemRefs *memRefs = NULL;
+    if(GetResourceManager()->OptimizeInitialState())
+      memRefs = GetResourceManager()->FindMemRefs(orig);
+    if(!memRefs)
+    {
+      // No information about the memory usage in the frame.
+      // Pessimistically assume the entire memory needs to be reset.
+      resetReq.update(0, initial.mem.size, eInitReq_Reset,
+                      [](InitReqType x, InitReqType y) -> InitReqType { return std::max(x, y); });
+    }
+    else
+    {
+      bool initialized = memRefs->initializedLiveRes == live;
```

Comment:
Do you think it would be better to pass in an extra parameter to `Apply_InitialState` saying whether or not this is the very first initialisation-time apply? I think we can trace it back to a specific call, the `if(IsLoading()) { ApplyInitialContents(); }` inside `WrappedVulkan::ContextReplayLog`, so we could pass in a flag there.

I'm not trying to lead the question either way - I could see an argument for both solutions so it's up to you.

Reply:
I played around with this a bit, and then put it back the way I had it originally, since this way feels a bit more robust against possible future changes. Right now it does seem to be the case that all the resources are initialized for the first time from that line, but if that were to change in the future, it could cause some strange behaviour that might be difficult to track down.

Reply:
Totally fair, that makes sense.